bears/general/AnnotationBear.py

```diff
@@ -28,104 +27,254 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
+            raise NoCloseError(annotation_start, _range)
+
+        return SourceRange.from_absolute_position(
+            filename,
+            AbsolutePosition(file, position),
+            AbsolutePosition(file, end_end)), end_end
+
+    @staticmethod
+    def get_singleline_strings(file,
+                               filename,
+                               text,
+                               string_start,
+                               string_end,
+                               position):
+        """
+        Get sourcerange of a single-line string and its end position.
+
+        :param file:
+            A tuple of strings, with each string being a line in the file.
+        :param filename:
+            The name of the file.
+        :param string_start:
+            The string which specifies how a string starts.
+        :param string_end:
+            The string which specifies how a string ends.
+        :position:
+            An integer identifying the position where the string started.
+        :return:
+            A SourceRange object identifying the range of the single-line
+            string and the end_position of the string as an integer.
+        """
+        end_position = (text.find(string_end, position + 1)
+                        + len(string_end) - 1)
+        newline = text.find("\n", position + 1)
+        if newline == -1:
+            newline = len(text)
+        if end_position == -1:
+            _range = SourceRange.from_absolute_position(
+                filename,
+                AbsolutePosition(file, position))
+            raise NoCloseError(string_start, _range)
+        if newline > end_position:
+            return SourceRange.from_absolute_position(
+                    filename,
+                    AbsolutePosition(file, position),
+                    AbsolutePosition(file, end_position)), end_position
+
+    @staticmethod
+    def get_singleline_comment(file, filename, text, comment, position):
+        """
+        Gets Sourcerange of a single-line comment where the start is the
+        start of comment and the end is the end of line.
+
+        :param file:
+            A tuple of strings, with each string being a line in the file.
+        :param filename:
+            The name of the file.
+        :param comment:
+            The string which specifies the comment.
+        :position:
+            An integer identifying the position where the string started.
+        :return:
+            A SourceRange object identifying the range of the single-line
+            comment and the end_position of the comment as an integer.
+        """
+        end_position = text.find("\n", position + 1)
+        if end_position == -1:
+            end_position = len(text) - 1
+        return SourceRange.from_absolute_position(
+                filename,
+                AbsolutePosition(file, position),
+                AbsolutePosition(file, end_position)), end_position
```

Comment:
that `end_position` is kinda hard to spot this way.

maybe return a touple (in parentheses) and indent this differently?
I'm thinking of something like:

```
return (SourceRange.from_absolute_position(filename,
                                           AbsolutePosition(file, position),
                                           AbsolutePosition(file, end_position)), 
        end_position)
```